src/libraries/icubmod/embObjBattery/embObjBattery.cpp

```diff
@@ -313,7 +428,7 @@ bool embObjBattery::getBatteryVoltage(double &voltage)
 
 bool embObjBattery::getBatteryCurrent(double &current)
 {
-    current = canBatteryData_.current_;
+    current = (canBatteryData_.current_ != 0) ? canBatteryData_.current_ : NAN;
```

Comment:
Since `current` is a float, here I'm not sure that the current is converted into int and then compared to 0. So I suggest comparing `current` with `0.0`, in order to avoid truncating values.